src/OrbitBase/include/OrbitBase/TaskGroup.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TASK_GROUP_H_
+#define ORBIT_BASE_TASK_GROUP_H_
+
+#include "OrbitBase/ThreadPool.h"
+
+namespace orbit_base {
+
+// Simple class used to parallelize a group of tasks. Add tasks to be run in parallel then call
+// "Wait()" to wait for all tasks to complete. Note that "Wait()" is called from the destructor if
+// it wasn't explicitly called. This class is not thread-safe. A TaskGroup object should be owned
+// and accessed by a single thread.
+//
+// Usage:
+//
+// void ProcessObjectsInParallel(std::vector<Object>& objects) {
+//   TaskGroup task_group(thread_pool_);
+//   for (Object& object : objects) {
+//     task_group.AddTask([&object]() { ProcessObject(object); });
+//   }
+// }
```

Comment:
I'm curious about the use-case. Why do you prefer this over `JoinFutures`? You could do something like this with `JoinFutures`:

```cpp
std::vector<Future<void>> futures;

for (Object& object : objects) {
  thread_pool->Schedule([&object]() { ProcessObject(object); });
}

orbit_base::JoinFutures(futures).Wait(); // Waits for all the futures to complete.
```

Reply:
I saw that pattern half-way into the implementation of TaskGroup. I think TaskGroup still has its place. I think it's more expressive, it's self-contained and you don't have to call "JoinFutures" explicitly, saving a line of code. Also, if we add the "IsFinished" method, then having a TaskGroup object is easier to use than the equivalent code with the std::vector<std::futures>.